synapse/storage/pusher.py

```diff
@@ -76,6 +76,25 @@ def r(txn):
         defer.returnValue(rows)
 
     @defer.inlineCallbacks
+    def get_pushers_by_app_user_id(self, user_id):
+        def r(txn):
+            sql = (
+                "SELECT * FROM pushers"
+                " WHERE user_name = ?"
+            )
+
+            txn.execute(sql, (user_id,))
+            rows = self.cursor_to_dict(txn)
+
+            return self._decode_pushers_rows(rows)
```

Comment:
Generally I prefer to do all CPU work on the main thread, to avoid taking out database threads. Might also be nice to use the `_simple_select_list` func.